### PR TITLE
Consistently use bazel test and --test_output=all on CI

### PR DESCRIPTION
**Pull request by lgeiger**
## What do these changes do?
This PR changes CI to consistently use `bazel test --test_output=all`. This will hopefully make sure that asserts are enabled during all of our tests.

## How Has This Been Tested?
CI

### CODE REVIEW

File: .github/workflows/unittests.yml [hunk 1]
```diff
@@ -48,9 +48,9 @@ jobs:
         run: ./configure.sh <<< $'n\n'
         shell: bash
       - name: "TF Lite Arm32: Cross-compile and run unit tests in qemu"
-        run: ./bazelisk run larq_compute_engine/tests:cc_tests_arm32_qemu --config=rpi3 --compilation_mode=fastbuild
+        run: ./bazelisk test larq_compute_engine/tests:cc_tests_arm32_qemu --config=rpi3 --compilation_mode=fastbuild --test_output=all
```
Comment (AdamHillier) [inline .github/workflows/unittests.yml:51]:
```suggestion
        run: ./bazelisk test larq_compute_engine/tests:cc_tests_arm32_qemu --config=rpi3 --test_output=all
```

File: .github/workflows/unittests.yml [hunk 2]
```diff
@@ -48,9 +48,9 @@ jobs:
...
       - name: "TF Lite Aarch64: Cross-compile and run unit tests in qemu"
-        run: ./bazelisk run larq_compute_engine/tests:cc_tests_aarch64_qemu --config=aarch64 --compilation_mode=fastbuild
+        run: ./bazelisk test larq_compute_engine/tests:cc_tests_aarch64_qemu --config=aarch64 --compilation_mode=fastbuild --test_output=all
```
Comment (AdamHillier) [inline .github/workflows/unittests.yml:53]:
```suggestion
        run: ./bazelisk test larq_compute_engine/tests:cc_tests_aarch64_qemu --config=aarch64 --test_output=all
```

File: .github/workflows/unittests.yml (hunk 1 above)
Comment (AdamHillier) [reply .github/workflows/unittests.yml:51]: (Test already sets fastbuild)

**Comment by lgeiger** [comment]
> This should have the very nice side benefit of speeding up the end2end test, because previously it was build with opt and so couldn't re-use the compilation of the MLIR filecheck, which was fastbuild.

That makes sense, I was wondering why it now has to rebuilt and is not streamed from the cache :+1:

**Comment by lgeiger** [comment]
@AdamHillier do you have an idea why the ARM tests timeout now?

**Comment by AdamHillier** [comment]
> @AdamHillier do you have an idea why the ARM tests timeout now?

Good question, I'm not sure. Previously they took a bit longer than this run did: https://github.com/larq/compute-engine/runs/567324056.

**Comment by lgeiger** [comment]
:+1: I'll increase the timeout if it fails after this run as well.

**Comment by Tombana** [comment]
In the int8 PR the ARM tests are 'fixed' because the qemu tests only run a small subset of parameters (this should probably be split into a separate PR from the int8 PR), so i think we can simply increase the timeout for now and fix it later.

**Comment by lgeiger** [comment]
> I wonder if bazel allows more time for "medium" tests when running -c opt or something like that.

I think it is because we used `bazel run` before, it probably doesn't check the timeout.

**Pull request merged**